lib/view_component/base.rb

```diff
@@ -308,7 +308,14 @@ def validate_collection_parameter!(validate_default: false)
         parameter = validate_default ? collection_parameter : provided_collection_parameter
 
         return unless parameter
-        return if instance_method(:initialize).parameters.map(&:last).include?(parameter)
+        return if initialize_parameters.map(&:last).include?(parameter)
+
+        # See https://github.com/github/view_component/pull/364
```

Comment:
Would you mind adding a brief description of the reason for this check as a code comment here, in place of a link to this PR?